### DISCUSSION TITLE
Control Transfers Fail for non HID Devices with Libusb

### DISCUSSION THREAD

**Discussion by EdAChu** [pyusb/pyusb#497] [Q&A] [answered]
Hi,

I'm running python on Windows 10 using libusb-1.0.26 and want
to be able to send control transfers to usb devices, but this
fails for devices that aren't HID devices, such as a hub or
camera. The functions that pyusb has that rely upon control
transfers, fail in the module libusb1.py, in the call to
_lib.libusb_open, in the class _DeviceHandle. The call to
libusb_open is a dll call of type _FuncPtr, in
'C:\\WINDOWS\\system32\\libusb-1.0.dll' (found in _lib._name)
and takes just two 64 bit parameters. The first is a device
ID (devid) and the second is the address of a pointer in which
the open call should return the address of a device handle
object (_DeviceHandle). If the call fails, an address isn't
returned and the function return value is negative (always
-12, LIBUSB_ERROR_NOT_SUPPORTED). It does raise the question
of why such a simple function as open fails for a non HID
device. The link:
https://github.com/libusb/libusb/wiki/Windows#How_to_use_libusb_on_Windows
does say: "If your target device is not HID, and your device
is not using WinUSB driver, you must install a driver before
you can communicate with it using libusb." There's a reference
to this in: https://github.com/pyusb/pyusb/discussions/419
Is there a way to fix this in Libusb without requiring a driver
change? While changing the driver can fix libusb_open, it
can break other applications that access the usb device.

**Comment by mcuee** [comment, marked as answer]
Please use cython-hidapi and not pyusb.
